### ISSUE TITLE
CMake-3.21.1-GCCcore-11.2.0.eb fails to install

### ISSUE THREAD

**Issue by VictorGoitea** [easybuilders/easybuild-easyconfigs#15705]
Hi 

Do someone understand what is going on that makes this installation to fail and how can be fixed?

Error Message:
```
FAILED: Installation ended unsuccessfully (build directory: /home/ucloud/.local/easybuild/build/CMake/3.21.1/GCCcore-11.2.0): build failed (first 300 chars): cmd " make install " exited with exit code 2 and output:
Consolidate compiler generated dependencies of target cmsys_c
[  1%] Built target cmsys_c
Consolidate compiler generated dependencies of target cmsysTestSharedForward
[  1%] Built target cmsysTestSharedForward
Consolidate compiler generated de (took 8 mins 52 secs)
== Results of the build can be found in the log file(s) /tmp/eb-_ub2josm/easybuild-CMake-3.21.1-20220617.074529.pUFXg.log
ERROR: Build of /opt/easybuild/easyconfigs/c/CMake/CMake-3.21.1-GCCcore-11.2.0.eb failed (err: 'build failed (first 300 chars): cmd " make install " exited with exit code 2 and output:\nConsolidate compiler generated dependencies of target cmsys_c\n[  1%] Built target cmsys_c\nConsolidate compiler generated dependencies of target cmsysTestSharedForward\n[  1%] Built target cmsysTestSharedForward\nConsolidate compiler generated de')
```

Error message in easybuild-CMake-3.21.1-20220617.074529.pUFXg.log file:
```
bin/cmake: error while loading shared libraries: libcrypto.so.1.1: cannot open shared object file: No such file or directory
make: *** [Makefile:144: install] Error 127
 (at easybuild/software/EasyBuild/4.5.5/lib/python3.10/site-packages/easybuild/tools/run.py:618 in parse_cmd_output)
== 2022-06-17 07:54:21,342 build_log.py:265 INFO ... (took 9 secs)
== 2022-06-17 07:54:21,343 filetools.py:1983 INFO Removing lock /work/easybuild/software/.locks/_work_easybuild_software_CMake_3.21.1-GCCcore-11.2.0.lock...
== 2022-06-17 07:54:21,450 filetools.py:382 INFO Path /work/easybuild/software/.locks/_work_easybuild_software_CMake_3.21.1-GCCcore-11.2.0.lock successfully removed.
== 2022-06-17 07:54:21,450 filetools.py:1987 INFO Lock removed: /work/easybuild/software/.locks/_work_easybuild_software_CMake_3.21.1-GCCcore-11.2.0.lock
== 2022-06-17 07:54:21,450 easyblock.py:4079 WARNING build failed (first 300 chars): cmd " make install " exited with exit code 2 and output:
Consolidate compiler generated dependencies of target cmsys_c
[  1%] Built target cmsys_c
Consolidate compiler generated dependencies of target cmsysTestSharedForward
[  1%] Built target cmsysTestSharedForward
Consolidate compiler generated de
== 2022-06-17 07:54:21,451 easyblock.py:318 INFO Closing log for application name CMake version 3.21.1
```

Since libcrypto.so.1.1 since to be part of  OpenSSL, I tried to install OpenSSL-1.1.1k-GCCcore-11.2.0.eb, but it also failed

```
eb --installpath=/work/easybuild OpenSSL-1.1.1k-GCCcore-11.2.0.eb -r

== FAILED: Installation ended unsuccessfully (build directory: /home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0): build failed (first 300 chars): cmd " make test " exited with exit code 2 and output:
make depend && make _tests
make[1]: Entering directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make[1]: Leaving directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k (took 5 mins 58 secs)
== Results of the build can be found in the log file(s) /tmp/eb-4gsqo33b/easybuild-OpenSSL-1.1.1k-20220617.094254.JAhIx.log
ERROR: Build of /opt/easybuild/easyconfigs/o/OpenSSL/OpenSSL-1.1.1k-GCCcore-11.2.0.eb failed (err: 'build failed (first 300 chars): cmd " make test " exited with exit code 2 and output:\nmake depend && make _tests\nmake[1]: Entering directory \'/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k\'\nmake[1]: Leaving directory \'/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k')


/tmp/eb-4gsqo33b/easybuild-OpenSSL-1.1.1k-20220617.094254.JAhIx.log


== 2022-06-17 09:45:06,011 run.py:233 INFO running cmd:  make test  
== 2022-06-17 09:48:53,324 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/software/EasyBuild/4.5.5/lib/python3.10/site-packages/easybuild/base/exceptions.py:124 in __init__): cmd " make test " exited with exit code 2 and output:
make depend && make _tests
make[1]: Entering directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make[1]: Leaving directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make[1]: Entering directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
( cd test; \
  mkdir -p test-runs; \
  SRCTOP=../. \
  BLDTOP=../. \
  RESULT_D=test-runs \
  PERL="/work/easybuild/software/Perl/5.34.0-GCCcore-11.2.0/bin/perl" \
  EXE_EXT= \
  OPENSSL_ENGINES=`cd .././engines 2>/dev/null && pwd` \
  OPENSSL_DEBUG_MEMORY=on \
    /work/easybuild/software/Perl/5.34.0-GCCcore-11.2.0/bin/perl .././test/run_tests.pl  )
../test/recipes/01-test_abort.t .................... ok
../test/recipes/01-test_sanity.t ................... ok
../test/recipes/01-test_symbol_presence.t .......... ok
../test/recipes/01-test_test.t ..................... ok
../test/recipes/02-test_errstr.t ................... ok
../test/recipes/02-test_internal_ctype.t ........... ok
../test/recipes/02-test_lhash.t .................... ok
../test/recipes/02-test_ordinals.t ................. ok
../test/recipes/02-test_stack.t .................... ok
../test/recipes/03-test_exdata.t ................... ok
../test/recipes/03-test_internal_asn1.t ............ ok
../test/recipes/03-test_internal_chacha.t .......... ok
../test/recipes/03-test_internal_curve448.t ........ ok
../test/recipes/03-test_internal_ec.t .............. ok
../test/recipes/03-test_internal_mdc2.t ............ ok
../test/recipes/03-test_internal_modes.t ........... ok
../test/recipes/03-test_internal_poly1305.t ........ ok
../test/recipes/03-test_internal_siphash.t ......... ok
../test/recipes/03-test_internal_sm2.t ............. ok
../test/recipes/03-test_internal_sm4.t ............. ok
../test/recipes/03-test_internal_ssl_cert_table.t .. ok
../test/recipes/03-test_internal_x509.t ............ ok
../test/recipes/03-test_ui.t ....................... ok
../test/recipes/04-test_asn1_decode.t .............. ok
../test/recipes/04-test_asn1_encode.t .............. ok
../test/recipes/04-test_asn1_string_table.t ........ ok
../test/recipes/04-test_bio_callback.t ............. ok
../test/recipes/04-test_bioprint.t ................. ok
../test/recipes/04-test_err.t ...................... ok
../test/recipes/04-test_pem.t ...................... ok
../test/recipes/05-test_bf.t ....................... ok
../test/recipes/05-test_cast.t ..................... ok
../test/recipes/05-test_cmac.t ..................... ok
../test/recipes/05-test_des.t ...................... ok
../test/recipes/05-test_hmac.t ..................... ok
../test/recipes/05-test_idea.t ..................... ok
../test/recipes/05-test_md2.t ...................... skipped: md2 is not supported by this OpenSSL build
../test/recipes/05-test_mdc2.t ..................... ok
../test/recipes/05-test_rand.t ..................... ok
../test/recipes/05-test_rc2.t ...................... ok
../test/recipes/05-test_rc4.t ...................... ok
../test/recipes/05-test_rc5.t ...................... skipped: rc5 is not supported by this OpenSSL build
../test/recipes/06-test-rdrand.t ................... ok
../test/recipes/10-test_bn.t ....................... ok
../test/recipes/10-test_exp.t ...................... ok
../test/recipes/15-test_dh.t ....................... ok
../test/recipes/15-test_dsa.t ...................... ok
../test/recipes/15-test_ec.t ....................... ok
../test/recipes/15-test_ecdsa.t .................... ok
../test/recipes/15-test_ecparam.t .................. ok
../test/recipes/15-test_genec.t .................... ok
../test/recipes/15-test_genrsa.t ................... ok
../test/recipes/15-test_mp_rsa.t ................... ok
../test/recipes/15-test_out_option.t ............... ok
../test/recipes/15-test_rsa.t ...................... ok
../test/recipes/15-test_rsapss.t ................... ok
../test/recipes/20-test_dgst.t ..................... ok
../test/recipes/20-test_enc.t ...................... ok
../test/recipes/20-test_enc_more.t ................. ok
../test/recipes/20-test_passwd.t ................... ok
../test/recipes/25-test_crl.t ...................... ok
../test/recipes/25-test_d2i.t ...................... ok
../test/recipes/25-test_pkcs7.t .................... ok
../test/recipes/25-test_req.t ...................... ok
../test/recipes/25-test_sid.t ...................... ok
../test/recipes/25-test_verify.t ................... ok
../test/recipes/25-test_x509.t ..................... ok
../test/recipes/30-test_afalg.t .................... ok
../test/recipes/30-test_engine.t ................... ok
../test/recipes/30-test_evp.t ...................... ok
../test/recipes/30-test_evp_extra.t ................ ok
../test/recipes/30-test_pbelu.t .................... ok
../test/recipes/30-test_pkey_meth.t ................ ok
../test/recipes/30-test_pkey_meth_kdf.t ............ ok
../test/recipes/40-test_rehash.t ................... ok
../test/recipes/60-test_x509_check_cert_pkey.t ..... ok
../test/recipes/60-test_x509_dup_cert.t ............ ok
../test/recipes/60-test_x509_store.t ............... ok
../test/recipes/60-test_x509_time.t ................ ok
../test/recipes/70-test_asyncio.t .................. ok
../test/recipes/70-test_bad_dtls.t ................. ok
../test/recipes/70-test_clienthello.t .............. ok
../test/recipes/70-test_comp.t ..................... ok
../test/recipes/70-test_key_share.t ................ ok
../test/recipes/70-test_packet.t ................... ok
../test/recipes/70-test_recordlen.t ................ ok
../test/recipes/70-test_renegotiation.t ............ ok
../test/recipes/70-test_servername.t ............... ok
../test/recipes/70-test_sslcbcpadding.t ............ ok
../test/recipes/70-test_sslcertstatus.t ............ ok
../test/recipes/70-test_sslextension.t ............. ok
../test/recipes/70-test_sslmessages.t .............. ok
../test/recipes/70-test_sslrecords.t ............... ok
../test/recipes/70-test_sslsessiontick.t ........... ok
../test/recipes/70-test_sslsigalgs.t ............... ok
../test/recipes/70-test_sslsignature.t ............. ok
../test/recipes/70-test_sslskewith0p.t ............. ok
../test/recipes/70-test_sslversions.t .............. ok
../test/recipes/70-test_sslvertol.t ................ ok
../test/recipes/70-test_tls13alerts.t .............. ok
../test/recipes/70-test_tls13cookie.t .............. ok
../test/recipes/70-test_tls13downgrade.t ........... ok
../test/recipes/70-test_tls13hrr.t ................. ok
../test/recipes/70-test_tls13kexmodes.t ............ ok
../test/recipes/70-test_tls13messages.t ............ ok
../test/recipes/70-test_tls13psk.t ................. ok
../test/recipes/70-test_tlsextms.t ................. ok
../test/recipes/70-test_verify_extra.t ............. ok
../test/recipes/70-test_wpacket.t .................. ok
../test/recipes/80-test_ca.t ....................... ok
../test/recipes/80-test_cipherbytes.t .............. ok
../test/recipes/80-test_cipherlist.t ............... ok
../test/recipes/80-test_ciphername.t ............... ok
../test/recipes/80-test_cms.t ...................... ok
../test/recipes/80-test_cmsapi.t ................... ok
../test/recipes/80-test_ct.t ....................... ok
../test/recipes/80-test_dane.t ..................... ok
../test/recipes/80-test_dtls.t ..................... ok
../test/recipes/80-test_dtls_mtu.t ................. ok
../test/recipes/80-test_dtlsv1listen.t ............. ok
../test/recipes/80-test_ocsp.t ..................... ok
../test/recipes/80-test_pkcs12.t ................... ok
../test/recipes/80-test_ssl_new.t .................. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/29 subtests 
../test/recipes/80-test_ssl_old.t .................. ok
../test/recipes/80-test_ssl_test_ctx.t ............. ok
../test/recipes/80-test_sslcorrupt.t ............... ok
../test/recipes/80-test_tsa.t ...................... ok
../test/recipes/80-test_x509aux.t .................. ok
../test/recipes/90-test_asn1_time.t ................ ok
../test/recipes/90-test_async.t .................... ok
../test/recipes/90-test_bio_enc.t .................. ok
../test/recipes/90-test_bio_memleak.t .............. ok
../test/recipes/90-test_constant_time.t ............ ok
../test/recipes/90-test_fatalerr.t ................. ok
../test/recipes/90-test_gmdiff.t ................... ok
../test/recipes/90-test_gost.t ..................... skipped: No test GOST engine found
../test/recipes/90-test_ige.t ...................... ok
../test/recipes/90-test_includes.t ................. ok
../test/recipes/90-test_memleak.t .................. ok
../test/recipes/90-test_overhead.t ................. skipped: Only supported in no-shared builds
../test/recipes/90-test_secmem.t ................... ok
../test/recipes/90-test_shlibload.t ................ ok
../test/recipes/90-test_srp.t ...................... ok
../test/recipes/90-test_sslapi.t ................... ok
../test/recipes/90-test_sslbuffers.t ............... ok
../test/recipes/90-test_store.t .................... ok
../test/recipes/90-test_sysdefault.t ............... ok
../test/recipes/90-test_threads.t .................. ok
../test/recipes/90-test_time_offset.t .............. ok
../test/recipes/90-test_tls13ccs.t ................. ok
../test/recipes/90-test_tls13encryption.t .......... ok
../test/recipes/90-test_tls13secrets.t ............. ok
../test/recipes/90-test_v3name.t ................... ok
../test/recipes/95-test_external_boringssl.t ....... skipped: No external tests in this configuration
../test/recipes/95-test_external_krb5.t ............ skipped: No external tests in this configuration
../test/recipes/95-test_external_pyca.t ............ skipped: No external tests in this configuration
../test/recipes/99-test_ecstress.t ................. ok
../test/recipes/99-test_fuzz.t ..................... ok

Test Summary Report
-------------------
../test/recipes/80-test_ssl_new.t                (Wstat: 256 Tests: 29 Failed: 1)
  Failed test:  12
  Non-zero exit status: 1
Files=158, Tests=2636, 227 wallclock secs ( 3.09 usr  0.51 sys + 94.29 cusr 143.53 csys = 241.42 CPU)
Result: FAIL
make[1]: *** [Makefile:208: _tests] Error 1
make[1]: Leaving directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make: *** [Makefile:205: tests] Error 2
 (at easybuild/software/EasyBuild/4.5.5/lib/python3.10/site-packages/easybuild/tools/run.py:618 in parse_cmd_output)
== 2022-06-17 09:48:53,325 build_log.py:265 INFO ... (took 3 mins 47 secs)
== 2022-06-17 09:48:53,326 filetools.py:1983 INFO Removing lock /work/easybuild/software/.locks/_work_easybuild_software_OpenSSL_1.1.1k-GCCcore-11.2.0.lock...
== 2022-06-17 09:48:53,327 filetools.py:382 INFO Path /work/easybuild/software/.locks/_work_easybuild_software_OpenSSL_1.1.1k-GCCcore-11.2.0.lock successfully removed.
== 2022-06-17 09:48:53,327 filetools.py:1987 INFO Lock removed: /work/easybuild/software/.locks/_work_easybuild_software_OpenSSL_1.1.1k-GCCcore-11.2.0.lock
== 2022-06-17 09:48:53,327 easyblock.py:4079 WARNING build failed (first 300 chars): cmd " make test " exited with exit code 2 and output:
make depend && make _tests
make[1]: Entering directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make[1]: Leaving directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k
== 2022-06-17 09:48:53,327 easyblock.py:318 INFO Closing log for application name OpenSSL version 1.1.1k
```

**Comment by VictorGoitea** [comment]
Your spider sense is in the right direction. I had tried to install OpenSSL-1.1.1k-GCCcore-11.2.0.eb but it also failed

`== 2022-06-17 09:45:06,011 run.py:233 INFO running cmd:  make test  
== 2022-06-17 09:48:53,324 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/software/EasyBuild/4.5.5/lib/python3.10/site-packages/easybuild/base/exceptions.py:124 in __init__): cmd " make test " exited with exit code 2 and output:
make depend && make _tests
make[1]: Entering directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make[1]: Leaving directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make[1]: Entering directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
( cd test; \
  mkdir -p test-runs; \
  SRCTOP=../. \
  BLDTOP=../. \
  RESULT_D=test-runs \
  PERL="/work/easybuild/software/Perl/5.34.0-GCCcore-11.2.0/bin/perl" \
  EXE_EXT= \
  OPENSSL_ENGINES=`cd .././engines 2>/dev/null && pwd` \
  OPENSSL_DEBUG_MEMORY=on \
    /work/easybuild/software/Perl/5.34.0-GCCcore-11.2.0/bin/perl .././test/run_tests.pl  )
../test/recipes/01-test_abort.t .................... ok
../test/recipes/01-test_sanity.t ................... ok
../test/recipes/01-test_symbol_presence.t .......... ok
../test/recipes/01-test_test.t ..................... ok
../test/recipes/02-test_errstr.t ................... ok
../test/recipes/02-test_internal_ctype.t ........... ok
../test/recipes/02-test_lhash.t .................... ok
../test/recipes/02-test_ordinals.t ................. ok
../test/recipes/02-test_stack.t .................... ok
../test/recipes/03-test_exdata.t ................... ok
../test/recipes/03-test_internal_asn1.t ............ ok
../test/recipes/03-test_internal_chacha.t .......... ok
../test/recipes/03-test_internal_curve448.t ........ ok
../test/recipes/03-test_internal_ec.t .............. ok
../test/recipes/03-test_internal_mdc2.t ............ ok
../test/recipes/03-test_internal_modes.t ........... ok
../test/recipes/03-test_internal_poly1305.t ........ ok
../test/recipes/03-test_internal_siphash.t ......... ok
../test/recipes/03-test_internal_sm2.t ............. ok
../test/recipes/03-test_internal_sm4.t ............. ok
../test/recipes/03-test_internal_ssl_cert_table.t .. ok
../test/recipes/03-test_internal_x509.t ............ ok
../test/recipes/03-test_ui.t ....................... ok
../test/recipes/04-test_asn1_decode.t .............. ok
../test/recipes/04-test_asn1_encode.t .............. ok
../test/recipes/04-test_asn1_string_table.t ........ ok
../test/recipes/04-test_bio_callback.t ............. ok
../test/recipes/04-test_bioprint.t ................. ok
../test/recipes/04-test_err.t ...................... ok
../test/recipes/04-test_pem.t ...................... ok
../test/recipes/05-test_bf.t ....................... ok
../test/recipes/05-test_cast.t ..................... ok
../test/recipes/05-test_cmac.t ..................... ok
../test/recipes/05-test_des.t ...................... ok
../test/recipes/05-test_hmac.t ..................... ok
../test/recipes/05-test_idea.t ..................... ok
../test/recipes/05-test_md2.t ...................... skipped: md2 is not supported by this OpenSSL build
../test/recipes/05-test_mdc2.t ..................... ok
../test/recipes/05-test_rand.t ..................... ok
../test/recipes/05-test_rc2.t ...................... ok
../test/recipes/05-test_rc4.t ...................... ok
../test/recipes/05-test_rc5.t ...................... skipped: rc5 is not supported by this OpenSSL build
../test/recipes/06-test-rdrand.t ................... ok
../test/recipes/10-test_bn.t ....................... ok
../test/recipes/10-test_exp.t ...................... ok
../test/recipes/15-test_dh.t ....................... ok
../test/recipes/15-test_dsa.t ...................... ok
../test/recipes/15-test_ec.t ....................... ok
../test/recipes/15-test_ecdsa.t .................... ok
../test/recipes/15-test_ecparam.t .................. ok
../test/recipes/15-test_genec.t .................... ok
../test/recipes/15-test_genrsa.t ................... ok
../test/recipes/15-test_mp_rsa.t ................... ok
../test/recipes/15-test_out_option.t ............... ok
../test/recipes/15-test_rsa.t ...................... ok
../test/recipes/15-test_rsapss.t ................... ok
../test/recipes/20-test_dgst.t ..................... ok
../test/recipes/20-test_enc.t ...................... ok
../test/recipes/20-test_enc_more.t ................. ok
../test/recipes/20-test_passwd.t ................... ok
../test/recipes/25-test_crl.t ...................... ok
../test/recipes/25-test_d2i.t ...................... ok
../test/recipes/25-test_pkcs7.t .................... ok
../test/recipes/25-test_req.t ...................... ok
../test/recipes/25-test_sid.t ...................... ok
../test/recipes/25-test_verify.t ................... ok
../test/recipes/25-test_x509.t ..................... ok
../test/recipes/30-test_afalg.t .................... ok
../test/recipes/30-test_engine.t ................... ok
../test/recipes/30-test_evp.t ...................... ok
../test/recipes/30-test_evp_extra.t ................ ok
../test/recipes/30-test_pbelu.t .................... ok
../test/recipes/30-test_pkey_meth.t ................ ok
../test/recipes/30-test_pkey_meth_kdf.t ............ ok
../test/recipes/40-test_rehash.t ................... ok
../test/recipes/60-test_x509_check_cert_pkey.t ..... ok
../test/recipes/60-test_x509_dup_cert.t ............ ok
../test/recipes/60-test_x509_store.t ............... ok
../test/recipes/60-test_x509_time.t ................ ok
../test/recipes/70-test_asyncio.t .................. ok
../test/recipes/70-test_bad_dtls.t ................. ok
../test/recipes/70-test_clienthello.t .............. ok
../test/recipes/70-test_comp.t ..................... ok
../test/recipes/70-test_key_share.t ................ ok
../test/recipes/70-test_packet.t ................... ok
../test/recipes/70-test_recordlen.t ................ ok
../test/recipes/70-test_renegotiation.t ............ ok
../test/recipes/70-test_servername.t ............... ok
../test/recipes/70-test_sslcbcpadding.t ............ ok
../test/recipes/70-test_sslcertstatus.t ............ ok
../test/recipes/70-test_sslextension.t ............. ok
../test/recipes/70-test_sslmessages.t .............. ok
../test/recipes/70-test_sslrecords.t ............... ok
../test/recipes/70-test_sslsessiontick.t ........... ok
../test/recipes/70-test_sslsigalgs.t ............... ok
../test/recipes/70-test_sslsignature.t ............. ok
../test/recipes/70-test_sslskewith0p.t ............. ok
../test/recipes/70-test_sslversions.t .............. ok
../test/recipes/70-test_sslvertol.t ................ ok
../test/recipes/70-test_tls13alerts.t .............. ok
../test/recipes/70-test_tls13cookie.t .............. ok
../test/recipes/70-test_tls13downgrade.t ........... ok
../test/recipes/70-test_tls13hrr.t ................. ok
../test/recipes/70-test_tls13kexmodes.t ............ ok
../test/recipes/70-test_tls13messages.t ............ ok
../test/recipes/70-test_tls13psk.t ................. ok
../test/recipes/70-test_tlsextms.t ................. ok
../test/recipes/70-test_verify_extra.t ............. ok
../test/recipes/70-test_wpacket.t .................. ok
../test/recipes/80-test_ca.t ....................... ok
../test/recipes/80-test_cipherbytes.t .............. ok
../test/recipes/80-test_cipherlist.t ............... ok
../test/recipes/80-test_ciphername.t ............... ok
../test/recipes/80-test_cms.t ...................... ok
../test/recipes/80-test_cmsapi.t ................... ok
../test/recipes/80-test_ct.t ....................... ok
../test/recipes/80-test_dane.t ..................... ok
../test/recipes/80-test_dtls.t ..................... ok
../test/recipes/80-test_dtls_mtu.t ................. ok
../test/recipes/80-test_dtlsv1listen.t ............. ok
../test/recipes/80-test_ocsp.t ..................... ok
../test/recipes/80-test_pkcs12.t ................... ok
../test/recipes/80-test_ssl_new.t .................. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/29 subtests 
../test/recipes/80-test_ssl_old.t .................. ok
../test/recipes/80-test_ssl_test_ctx.t ............. ok
../test/recipes/80-test_sslcorrupt.t ............... ok
../test/recipes/80-test_tsa.t ...................... ok
../test/recipes/80-test_x509aux.t .................. ok
../test/recipes/90-test_asn1_time.t ................ ok
../test/recipes/90-test_async.t .................... ok
../test/recipes/90-test_bio_enc.t .................. ok
../test/recipes/90-test_bio_memleak.t .............. ok
../test/recipes/90-test_constant_time.t ............ ok
../test/recipes/90-test_fatalerr.t ................. ok
../test/recipes/90-test_gmdiff.t ................... ok
../test/recipes/90-test_gost.t ..................... skipped: No test GOST engine found
../test/recipes/90-test_ige.t ...................... ok
../test/recipes/90-test_includes.t ................. ok
../test/recipes/90-test_memleak.t .................. ok
../test/recipes/90-test_overhead.t ................. skipped: Only supported in no-shared builds
../test/recipes/90-test_secmem.t ................... ok
../test/recipes/90-test_shlibload.t ................ ok
../test/recipes/90-test_srp.t ...................... ok
../test/recipes/90-test_sslapi.t ................... ok
../test/recipes/90-test_sslbuffers.t ............... ok
../test/recipes/90-test_store.t .................... ok
../test/recipes/90-test_sysdefault.t ............... ok
../test/recipes/90-test_threads.t .................. ok
../test/recipes/90-test_time_offset.t .............. ok
../test/recipes/90-test_tls13ccs.t ................. ok
../test/recipes/90-test_tls13encryption.t .......... ok
../test/recipes/90-test_tls13secrets.t ............. ok
../test/recipes/90-test_v3name.t ................... ok
../test/recipes/95-test_external_boringssl.t ....... skipped: No external tests in this configuration
../test/recipes/95-test_external_krb5.t ............ skipped: No external tests in this configuration
../test/recipes/95-test_external_pyca.t ............ skipped: No external tests in this configuration
../test/recipes/99-test_ecstress.t ................. ok
../test/recipes/99-test_fuzz.t ..................... ok

Test Summary Report
-------------------
../test/recipes/80-test_ssl_new.t                (Wstat: 256 Tests: 29 Failed: 1)
  Failed test:  12
  Non-zero exit status: 1
Files=158, Tests=2636, 227 wallclock secs ( 3.09 usr  0.51 sys + 94.29 cusr 143.53 csys = 241.42 CPU)
Result: FAIL
make[1]: *** [Makefile:208: _tests] Error 1
make[1]: Leaving directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make: *** [Makefile:205: tests] Error 2
 (at easybuild/software/EasyBuild/4.5.5/lib/python3.10/site-packages/easybuild/tools/run.py:618 in parse_cmd_output)
== 2022-06-17 09:48:53,325 build_log.py:265 INFO ... (took 3 mins 47 secs)
== 2022-06-17 09:48:53,326 filetools.py:1983 INFO Removing lock /work/easybuild/software/.locks/_work_easybuild_software_OpenSSL_1.1.1k-GCCcore-11.2.0.lock...
== 2022-06-17 09:48:53,327 filetools.py:382 INFO Path /work/easybuild/software/.locks/_work_easybuild_software_OpenSSL_1.1.1k-GCCcore-11.2.0.lock successfully removed.
== 2022-06-17 09:48:53,327 filetools.py:1987 INFO Lock removed: /work/easybuild/software/.locks/_work_easybuild_software_OpenSSL_1.1.1k-GCCcore-11.2.0.lock
== 2022-06-17 09:48:53,327 easyblock.py:4079 WARNING build failed (first 300 chars): cmd " make test " exited with exit code 2 and output:
make depend && make _tests
make[1]: Entering directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k'
make[1]: Leaving directory '/home/ucloud/.local/easybuild/build/OpenSSL/1.1.1k/GCCcore-11.2.0/openssl-1.1.1k
== 2022-06-17 09:48:53,327 easyblock.py:318 INFO Closing log for application name OpenSSL version 1.1.1k
`
However, I had OpenSSL/1.1 already installed (I think via OpenSSL-1.1.eb). I checked in `lib` and you are right again, there are files symlink to the system binaries/libraries. This must be related to our recent Linux Ubuntu OS version upgrade.

How would it be possible to fix OpenSSL/1.1 to make it independent of the symlinks? Should I rebuild OpenSSL/1.1? Should the optional argument force be necessary as well? Afterwards, would it be necessary to rebuild the GCCcore-11.2.0 or just CMake-3.22?

Thanks in advance

**Comment by VictorGoitea** [comment]
Okay, I managed to install CMake-3.21.1 after rebuilding OpenSSL/1.1 via OpenSSL-1.1.eb. I have also tried to install OpenSSL-1.1.1k-GCCcore-11.2.0.eb but it kept failing at the test level, even after rebuilding all its dependencies. It keeps failing at **80-test_ssl_new.t** and I couldn't find a convincing solution to pass that test. Do you know how to solve this issue? I have used `--ignore-test-failure` to make the installation go through, though I understand the ideal solution should solve the failed test instead.

**Comment by Micket** [comment]
>  This must be related to our recent Linux Ubuntu OS version upgrade.

Well, that can break pretty much everything in the software tree if it's a significant update. 
I'm rebuilding from scratch now when I'm updating a cluster from CentOS7 to Rocky 8.

> How would it be possible to fix OpenSSL/1.1 to make it independent of the symlinks?

It's not broken, we very much intentionally did this, because we don't want a fixed version of security sensitive software.
The easyconfig has comments instruction you how to build a fixed version instead if you want to.

>  Should I rebuild OpenSSL/1.1? Should the optional argument force be necessary as well?

Yes. --rebuild flag should suffice.

> Afterwards, would it be necessary to rebuild the GCCcore-11.2.0 or just CMake-3.22?

Maybe, depends on what the OS update changed. It applies to all software.
I'm currently switching a cluster from CentOS7 to Rocky8 and trying to salvage anything from the old module tree is completely hopeless. I wipe all and start a new module tree.

--------

You don't want to use the `OpenSSL-1.1.1*-*` easyconfigs we have.. they are there.. well i'm not sure why we still have them. I recommend you don't use them.
Not that it should be broken, but, it's not an easyconfig that any of us devs probably have installed, so, i think we have to pick our battles here.
